.github/workflows/code.yml

```diff
@@ -20,21 +20,29 @@ jobs:
       - name: Setup python
         uses: actions/setup-python@v4
         with:
-          python-version: "3.10"
+          python-version: "3.x"
 
       - name: Lint
         run: |
-          touch requirements_dev.txt requirements.txt
-          pip install -r requirements.txt -r requirements_dev.txt -e .[dev]
+          touch requirements-lint.txt requirements.txt
+          pip install -r requirements.txt -r requirements-lint.txt -e .[dev]
```

Comment:
Only problem is that then the artifacts will not have been sed'd so can't just be copied into the source repo... Could we run the script in the container too?

Reply:
The container looks really neat and does not do anything special at the moment. I don't want to dirty it with extra knowledge about files in the gihub actions folders.

Plus this would be moving the requirements.txt generation into the container (not terrible) and also the creation of the lockfiles folder in the host filesystem from inside the container (really terrible).

Reply:
Isnt it more important that a release has requirements files than a build? Isn't the point that you'll need these when you come back to work from a previous release? 

Also the fix if you really do want to use one from an artefact is trivial.

Reply:
oops - the lint lockfile has
```
-e git+https://github.com/DiamondLightSource/python3-pip-skeleton-cli@a3b89ae5fcdb888b922884d214a0a6c59659671b#egg=python3_pip_skeleton
```
which the sed does not pick up

Reply:
adding --exclude-editable to pip freeze ...

Reply:
Doesn't the sed just comment out the first line?

Reply:
> Isnt it more important that a release has requirements files than a build? Isn't the point that you'll need these when you come back to work from a previous release?
> 
> Also the fix if you really do want to use one from an artefact is trivial.

That's true, might as well leave the sed in the release section then